routing/routing.go

```diff
@@ -0,0 +1,73 @@
+package routing
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"text/template"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	eth1FileName = "/etc/systemd/network/10-eth1.network"
+)
+
+type params struct {
+	ENIAddress    string
+	ENIGateway    string
+	ENISubnetSize int
+}
+
+func ConfigureNetworkRoutingForENI(eniIP string, eniSubnet *net.IPNet) error {
+	p := params{
+		ENIAddress:    eniIP,
+		ENIGateway:    eniGateway(eniSubnet),
+		ENISubnetSize: eniSubnetSize(eniSubnet),
+	}
+
+	err := renderRoutingNetworkdFile(p)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func renderRoutingNetworkdFile(p params) error {
+	var buff bytes.Buffer
+	t := template.Must(template.New("routing").Parse(networkRoutingTemplate))
+
+	err := t.Execute(&buff, p)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = ioutil.WriteFile(eth1FileName, buff.Bytes(), 0644)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func eniGateway(ipNet *net.IPNet) string {
+	// https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html
+	gatewayAddressIP := dupIP(ipNet.IP)
+	gatewayAddressIP.To4()
```

Comment:
I think this is wrong here in case `ipNet` would ever be IPv6 network. See https://golang.org/pkg/net/#IP.To4 and https://golang.org/src/net/ip.go?s=5275:5296#L179

Method receiver is not modified here, but if you need this, then you must take the return value and handle the potential error (i.e. `nil`) as well.

Reply:
Yeah true, I copy-pasted this from somewhere.